service/observiq-otel-collector

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
+# observIQ OTEL daemon
+# chkconfig: 2345 99 05
+# description: observIQ's distribution of the OpenTelemetry collector
+# processname: observiq-otel-collector
+# pidfile: /var/run/observiq-otel-collector.pid
+
+# Source function library.
+if [ -e /etc/init.d/functions ]; then
+  STATUS=true
+  . /etc/init.d/functions
+fi
+
+if [ -e /lib/lsb/init-functions ]; then
+  PROC=true
+  . /lib/lsb/init-functions
+fi
+
+# Pull in sysconfig settings
+[ -f /etc/sysconfig/observiq-otel-collector ] && . /etc/sysconfig/observiq-otel-collector
+
+PROGRAM=observiq-otel-collector
+LOCKFILE=/var/lock/$PROGRAM
+OIQ_OTEL_COLLECTOR_HOME=/opt/observiq-otel-collector
+OIQ_OTEL_COLLECTOR_STORAGE=/opt/observiq-otel-collector/storage
+HTTP_PROXY=
+HTTPS_PROXY=
+PIDFILE=/var/run/observiq-otel-collector.pid
+DEBUG=false
+RETVAL=0
+start() {
+  if [ -f $PIDFILE ]; then
+    PID=$(cat $PIDFILE)
+    echo " * $PROGRAM already running: $PID"
+    RETVAL=2
+  else
+    echo " * Starting $PROGRAM"
+    if [ $DEBUG ]; then
+      COMMAND='"/opt/observiq-otel-collector/observiq-otel-collector --config config.yaml" 1> /dev/null 2>> $OIQ_OTEL_COLLECTOR_HOME/log/error.log'
+    else
+      COMMAND='"/opt/observiq-otel-collector/observiq-otel-collector --config config.yaml" 1>> $OIQ_OTEL_COLLECTOR_HOME/log/error.log 2>&1'
+    fi
+    export OIQ_OTEL_COLLECTOR_HOME
+    export OIQ_OTEL_COLLECTOR_STORAGE
+    export HTTP_PROXY
+    export HTTPS_PROXY
+    if [ -n "root" ]; then
```

Comment:
Do we need this condition? Is there a scenario where the service would not be executed as the root user?

Also, how do we support running the collector as the `observiq-otel-collector` user (which is created and managed by the linux packages)? This does not need to be solved right away, just curious as we do have a solution for systemd.

```
    if [ -n "root" ]; then
             ^--^ SC2157: Argument to -n is always true due to literal strings.
```

Reply:
I've addressed all shellcheck feedback (interesting some of them, since I copied this from an existing RHEL created script).

Additionally added the support to specify a user in place of the check you asked about, which was a placeholder I had forgotten.